### ISSUE TITLE
Rendering v2

### ISSUE THREAD

**Issue by jonyrock** [NatelEnergy/grafana-discrete-panel#31]
This is how my rendering looks like:
```
  onRender() {
    if(this.data == null || !(this.context)) {
      return;
    }
    this._updateRenderDimensions();
    this._updateSelectionMatrix();
    this._updateCanvasSize();
    this._renderRects();
    this._renderLabels();
    this._renderSelection();
    this._renderCrosshair();
  }
```

I render everything by layers, keeping `selectionMatrix` - matrix bitmap of selected rects and additional info: https://github.com/CorpGlory/grafana-discrete-panel/blob/master/src/module.js#L175

It is remade rendering which allows me to make many new features on top of it and makes code mush cleaner. 

This rendering also makes appearance different: different labels positions, styles of rects, selection, and crosshair.

**Comment by ryantxu** [comment]
this looks fine -- can you make a branch/pull request so it is easier to see the diff?

**Comment by jonyrock** [comment]
@ryantxu the thing is that I can't make small diffs, I think it would be better if I make a small video where I describe how the new version of code works and then we would make a strategy how to go forward.

**Comment by ryantxu** [comment]
You can make a large diff... essentially the thing you think makes sense to merge

**Comment by ryantxu** [comment]
@jonyrock -- anything you want to to look at?  The approach in CorpGlory/grafana-discrete-panel looks fine, but I think looking at a mergable branch would be the best way to move forward.

**Comment by jonyrock** [comment]
@rozetko let's elaborate on this

**Comment by ryantxu** [comment]
resolved with #42
